cpp/example_code/s3/README.md

```diff
@@ -1,54 +1,67 @@
+- [Getting and setting the access control list (ACL) for an S3 bucket](./get_put_bucket_acl.cpp) (GetBucketAcl, PutBucketAcl)
+- [Getting and setting the access control list (ACL) for an object in an S3 bucket](./get_put_object_acl.cpp) (GetObjectAcl, PutObjectAcl)
+- [Getting configuration of an S3 bucket configured for static website hosting](./get_website_config.cpp) (GetBucketWebsite)
+- [Listing all S3 buckets](./list_buckets.cpp) (GetBuckets)
+- [Listing all objects in an S3 bucket](./list_objects.cpp) (ListObjects)
+- [Making requests to S3 across AWS Regions by specifying aws-global as the AWS Region](./list_objects_with_aws_global_region.cpp)
+- [Adding a bucket policy (permission to access resources) to an S3 bucket](./put_bucket_policy.cpp) (PutBucketPolicy)
+- [Uploading an object to an S3 bucket](./put_object.cpp) (PutObject)
+- [Uploading an object to an S3 bucket (asynchronously)](./put_object_async.cpp) (PutObjectAsync)
+- [Uploading an object to an S3 bucket (using a memory buffer instead of local disk copy)](./put_object_buffer.cpp) (PutObject)
+- [Configuring an S3 bucket for static website hosting](./put_website_config.cpp) (PutBucketWebsite)
+
+### Scenarios
+- [Creating, listing, and deleting S3 buckets](./s3_getting_started_scenario.cpp)
+- [Finding, creating, and deleting an S3 bucket in a sequence](./s3-demo.cpp)
+
+## Running the examples
 Before using the code examples, first complete the installation and setup steps of [Getting Started](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html) in the AWS SDK for C++ Developer Guide.
-The Getting Started section covers how to obtain and build the SDK, and how to build your own code utilizing the SDK with a sample “Hello World”-style application. 
+The Getting Started section covers how to get and build the SDK, and how to build your own code by using the SDK with a sample “Hello World”-style application. 
 
 Next, see [Getting started with the AWS SDK for C++ code examples](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started-code-examples.html) for information on the structure of the code examples, building, and running the examples.
 
-## Additional Information
-See [Amazon S3 code examples](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/examples-s3.html) in the AWS SDK for C++ Developer Guide for additional information on using the Amazon S3 service with the SDK.
+## Tests
+⚠️ Running the tests might result in charges to your AWS account.
+
+```sh
+   cd <BUILD_DIR>
+   cmake <path-to-root-of-this-source-code> -DBUILD_TESTS=ON
+   make
+   ctest 
+   ./gtests/s3_gtest 
+```   
 
-## Resources
+## Additional resources
+- [Amazon Simple Storage Service Documentation](https://docs.aws.amazon.com/s3/index.html)
+- [Amazon S3 code examples](https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/examples-s3.html)
 - [AWS SDK for C++ Documentation](https://docs.aws.amazon.com/sdk-for-cpp/index.html) 
```

Comment:
(Optional) Ideally, this links to the S3 section of the C++ SDK reference, but fine to leave it for now.